Just/Just.swift

```
@@ -901,7 +901,20 @@ public final class HTTP: NSObject, NSURLSessionDelegate, JustAdaptor {
                         task.resume()
                     }
                     if isSync {
-                        dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
+                        // Express one second in nanoseconds.
+                        let NSEC: Int64 = 1000000000
+
+                        if let tint = timeout {
+                            // See Issue #30 for an explanation of why this is needed.
+                            let how_long = dispatch_time(DISPATCH_TIME_NOW, Int64(tint) * NSEC)
+                            let timed_out = dispatch_semaphore_wait(semaphore, how_long)
+                            if timed_out != 0 {
```

We can ignore the error code for now.